pkg/registry/pod/rest.go

```diff
@@ -81,9 +81,28 @@ func (podStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) fiel
 	return append(errorList, validation.ValidatePodUpdate(obj.(*api.Pod), old.(*api.Pod))...)
 }
 
-// CheckGracefulDelete allows a pod to be gracefully deleted.
+// CheckGracefulDelete allows a pod to be gracefully deleted. It updates the DeleteOptions to
+// reflect the desired grace value.
 func (podStrategy) CheckGracefulDelete(obj runtime.Object, options *api.DeleteOptions) bool {
-	return false
+	if options == nil {
+		return false
+	}
+	pod := obj.(*api.Pod)
+	// use the user provided value
+	if options.GracePeriodSeconds != nil {
+		period := int64(*options.GracePeriodSeconds)
+		// if the pod is not currently being deleted, or the value isn't set, or the user value is less than the stored value
+		if pod.Spec.TerminationGracePeriodSeconds == nil || pod.DeletionTimestamp == nil || period < *pod.Spec.TerminationGracePeriodSeconds {
+			pod.Spec.TerminationGracePeriodSeconds = &period
```

Comment:
The main problem I have with changing TerminationGracePeriodSeconds is that it would generate a conflict with user-provided configuration, in violation of the API rules:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/api-conventions.md#late-initialization

Also, where are we going to record GracePeriodSeconds for other object kinds?

I view TerminationGracePeriodSeconds as the user-specified default for use by controllers, when the user can't directly provide a more specific GracePeriodSeconds upon deletion.

If we need another field for recording the (relative) GracePeriodSeconds, we should add one. We could keep DeletionTimestamp, or replace it. I imagine we'll also want an LastUpdatedTimestamp, which might be a sufficient replacement for debugging purposes (not that we have to implement that now).